src/overlays/actors/ovl_Bg_Breakwall/z_bg_breakwall.c

```diff
@@ -53,21 +66,268 @@ static ColliderQuadInit D_80870760 = {
+void BgBreakwall_SetupAction(BgBreakwall* this, BgBreakwallActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void BgBreakwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgBreakwall* this = THIS;
+    s32 pad;
+    s32 wallType = ((this->dyna.actor.params >> 13) & 3) & 0xFF;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+    this->bombableWallDList = sBombableWallInfo[wallType].dList;
+    this->colType = sBombableWallInfo[wallType].colType;
+
+    if (this->colType == 1) {
+        this->dyna.actor.world.rot.x = 0x4000;
+    }
+
+    if (this->bombableWallDList != NULL) {
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F)) {
+            Actor_Kill(&this->dyna.actor);
+            return;
+        }
+
+        ActorShape_Init(&this->dyna.actor.shape, 0.0f, NULL, 0.0f);
+        Collider_InitQuad(globalCtx, &this->collider);
+        Collider_SetQuad(globalCtx, &this->collider, &this->dyna.actor, &sQuadInit);
+    } else {
+        this->dyna.actor.world.pos.y -= 40.0f;
+    }
+
+    this->bankIndex = (wallType >= BWALL_KD_FLOOR) ? Object_GetIndex(&globalCtx->objectCtx, OBJECT_KINGDODONGO)
+                                                   : Object_GetIndex(&globalCtx->objectCtx, OBJECT_BWALL);
+
+    if (this->bankIndex < 0) {
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        BgBreakwall_SetupAction(this, BgBreakwall_SetupWait);
+    }
+}
+
+void BgBreakwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgBreakwall* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+/**
+ * Spawns fragments using ACTOR_EN_A_OBJ whenever the wall or floor is exploded.
+ * Although it is not ever checked, it will also return the instance of the actor which it spawns.
+ */
+Actor* BgBreakwall_SpawnFragments(GlobalContext* globalCtx, BgBreakwall* this, Vec3f* pos, f32 velocity, f32 scaleY,
+                                  f32 scaleX, s32 count, f32 accel) {
+    Actor* actor;
+    Vec3f actorPos;
+    s32 k;
+    s32 j;
+    s32 i;
+    s16 angle1;
+    s16 angle2 = 0;
+    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f }; // unused
+    Vec3s actorRotList[] = { { 0, 0, 0 }, { 0, 0, 0x4000 }, { 0, 0, -0x4000 }, { 0, 0, 0 } };
+    Vec3f actorScaleList[] = {
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+        { 0.004f, 0.004f, 0.004f },
+    };
+    Vec3f actorPosList[][4] = {
+        { { 40.0f, 15.0f, 0.0f }, { 30.0f, 57.0f, 0.0f }, { 50.0f, 57.0f, 0.0f }, { 40.0f, 70.0f, 0.0f } },
+        { { 55.0f, -15.0f, 0.0f }, { 30.0f, -32.0f, 0.0f }, { 50.0f, -32.0f, 0.0f }, { 20.0f, -10.0f, 0.0f } },
+        { { -40.0f, 14.0f, 0.0f }, { -50.0f, 57.0f, 0.0f }, { -30.0f, 57.0f, 0.0f }, { -40.0f, 70.0f, 0.0f } },
+        { { -55.0f, -15.0f, 0.0f }, { -55.0f, -32.0f, 0.0f }, { -30.0f, -32.0f, 0.0f }, { -20.0f, -10.0f, 0.0f } },
+    };
```

Comment:
Wow, got enough inlined data?

Reply:
I mean it all has to be inlined, can't really do anything about it :P